### PR TITLE
Added cart features

### LINKED ISSUE #1657: Implement complete cart functionalities.
Currently in the website the cart is not working.

I can do the following changes.

1. Add items to cart when user clicks on add to cart button. If item already exists quantity will be increased.
2. In the cart page selected items will be listed. 
3. Users can update (+/-) the quantity.
4. Users can remove an item from the cart.

There are 2 options to implement cart feature. One is using local storage. Other one is storing in server.

I will be okay with both options.  Please let me know, which method you would like to proceed with this approach or if you have any further suggestions.

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**
## Fixes Issue

This pr fixes #1657 

## Changes proposed

The following changes have been implemented:

1. Add to Cart Functionality:

When users click the "Add to Cart" button, items are added to the cart.
If the item already exists in the cart, its quantity is increased.
Cart Page Enhancements:

2. Selected items are now listed on the cart page.

Users can update the quantity of items (+/-).
Users have the option to remove items from the cart.
User can clear the cart as well.

3. Guest Cart Experience:

Users can add items to the cart without logging in.
Cart items are stored in the user's browser to enhance the user experience.
Users must log in before proceeding to checkout.

## Screenshots

https://github.com/codervivek5/VigyBag/assets/139672976/50507495-af04-4194-ad60-1f4450ace1b6

https://github.com/codervivek5/VigyBag/assets/139672976/092d34d2-2804-42ed-bddf-f75208baceb7

Kindly let me know if any changes required!

## Summary by CodeRabbit

- **New Features**
	- Introduced Redux for state management, including a new shopping cart slice.
	- Added new custom hook `UseCart` for cart operations.
	- Always display the cart icon in the navbar regardless of login state.

- **Enhancements**
	- Improved shopping cart interface with new components for empty cart display, checkout, and login prompts.
	- Enhanced "Add to Cart" functionality with alerts and better state management.

- **UI/UX Improvements**
	- Updated empty cart display with new styling and content adjustments.
	- Restructured product grid and cart pages for better user interaction.

### CODE REVIEW

File: src/pages/Order/Cart.jsx [hunk 1]
```diff
@@ -50,9 +50,9 @@ const CartItem = ({
       trigger="hover"
       colors="primary:#ff0000"
       style={{ width: "30px", height: "30px", cursor: "pointer" }}
-      onClick={() => onRemove(id)}
+      onClick={() => onupdate(product, -1 * product.quantity)}
       onKeyUp={(e) => {
-        if (e.key === "Enter") onRemove(id);
+        if (e.key === "Enter") onupdate(product, -1 * product.quantity);
```
[the remark this answers is not in the document]
Comment (codervivek5) [reply src/pages/Order/Cart.jsx]: update this change

File: src/hooks/UseCart.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+function UseCart() {
+
+    const addToCart = (product, quantity) => {
```
[the remark this answers is not in the document]
Comment (codervivek5) [reply src/hooks/UseCart.jsx]: update this change then let me know....I will merge it

**Comment by codervivek5** [comment]
@DharshiBalasubramaniyam if you are aware of the redux tool kit then use that for managing these types of actions and functionality.

**Comment by SadafKausar2025** [comment]
hello @DharshiBalasubramaniyam 
when the cart is becoming empty can you please add the Emptycart page there which is already in the vigyBag
src->pages->order->Emptycart

**Comment by codervivek5** [comment]
Can you share new changes

**Pull request closed without merging**